src/fsharp/IlxGen.fs

```diff
@@ -6390,20 +6339,27 @@ and GenTypeDef cenv mgbuf lazyInitInfo eenv m (tycon:Tycon) =
                      | TRecdRepr _ when not useGenuineField -> [ cenv.g.DebuggerBrowsableNeverAttribute ] // hide fields in records in debug display
                      | _ -> [] // don't hide fields in classes in debug display
 
-                  yield
+                  let access = ComputeMemberAccess isFieldHidden
+                  let literalValue = Option.map (GenFieldInit m) fspec.LiteralValue
+                  
+                  let fdef =
                       { Name          = ilFieldName
                         Type          = ilPropType
-                        IsStatic      = isStatic
-                        Access        = ComputeMemberAccess isFieldHidden
+                        Attributes    = enum 0
                         Data          = None 
-                        LiteralValue  = Option.map (GenFieldInit m) fspec.LiteralValue
+                        LiteralValue  = literalValue
                         Offset        = ilFieldOffset
-                        IsSpecialName = (ilFieldName="value__" && tycon.IsEnumTycon)
                         Marshal       = ilFieldMarshal
-                        NotSerialized = ilNotSerialized 
-                        IsInitOnly    = false  
-                        IsLiteral     = fspec.LiteralValue.IsSome 
                         CustomAttrs   = mkILCustomAttrs (GenAttrs cenv eenv fattribs @ extraAttribs) } 
+                  let fdef = 
+                    fdef.WithAccess(access)
```

Comment:
this is generating a lot new intermediate records - is that really an improvement?

Reply:
@forki I don't think it matters here, this is just not a hotspot in allocation.  Intermediate garbage gets collected fast enough

That said, it's possible the F# optimizer would cope with inlining all the With* methods and would reduce the allocations down to one, though we'd need to check.